From 05612ab96cccaf151f910941eb56108ace64db5b Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
Date: Fri, 22 Feb 2019 07:52:09 +0900
Subject: [PATCH] Unzip doesn't handle large zip files well and crashes:

"This only happens if you have more then 16k entries and when one of
the 16k entry infos is reused it happend to be previously used for
a symlink entry."

This patch is taken from Fedora (https://bugzilla.redhat.com/show_bug.cgi?id=972427)

Upstream-Status: Pending (upstream is dead)
Signed-off-by: Ross Burton <ross.burton@intel.com>
Signed-off-by: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
---
 process.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/process.c b/process.c
index e4f2405..b1c548f 100644
--- a/process.c
+++ b/process.c
@@ -1764,6 +1764,12 @@ int process_cdir_file_hdr(__G)    /* return PK-type error code */
     G.pInfo->symlink = 0;
 #endif
 
+#ifdef SYMLINKS
+    /* Initialize the symlink flag, may be set by the platform-specific
+       mapattr function.  */
+    G.pInfo->symlink = 0;
+#endif 
+
     return PK_COOL;
 
 } /* end function process_cdir_file_hdr() */
-- 
2.20.1

